src/coconut/data/macros/Models.hx

```diff
@@ -95,10 +95,16 @@ class Models {
 
     return macro @:pos(e.pos) {
       var ret = $e();
-      ret.handle(function (o) switch o {
-        case Success(v): __cocoupdate(v);
-        case _:
+      var handled = false;
+      var link = ret.handle(function (o) {
+        handled = true;
+        __coco_transitionLink.set(null);
+        switch o {
+          case Success(v): __cocoupdate(v);
+          case Failure(e): errorTrigger.trigger(e);
+        }
       });
+      if(!handled) __coco_transitionLink.set(link);
```

Comment:
This should be: 

```haxe
if (!handled) __coco_transitionLink.set(function () { 
  link.dissolve(); 
  if (transitionLink == link) __coco_transitionLink.set(null);
});
```

Otherwise dissolving the link merely unsubscribes the handler, but doesn't reset `transitionLink` to `null`.

Reply:
Thanks for pointing out!
But there is another problem `transitionLink == link` won't hold because the state now holds the anonymous function.

Reply:
Storing it in a temp var should work